akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -995,25 +995,30 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
       "not limit response streams even when the client send a SETTINGS_MAX_CONCURRENT_STREAMS" in new TestSetup(
         Setting(SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, 1)
       ) with RequestResponseProbes {
+        def openStream(streamId: Int) = sendHEADERS(streamId, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
+        def closeStream(streamId: Int) = responseOut.sendNext(HPackSpecExamples.FirstResponse.addAttribute(Http2.streamId, streamId))
+
         val request = HttpRequest(uri = "https://www.example.com/")
+
         // client set SETTINGS_MAX_CONCURRENT_STREAMS to 1 so an attempt from the server to open more streams
         // should fail. But as long as the outgoing streams are a result of client-initiated communication
         // they should succeed.
-        sendHEADERS(1, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
-        sendHEADERS(3, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
-        sendHEADERS(5, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
-        sendHEADERS(7, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
+        openStream(1)
```

Comment:
I kindof liked the explicit sending of request frames and response objects, makes it easier to see who is triggering the opening/closing. I'm OK with this as well, though.

Reply:
I agree with @raboof, but I think it's fine in the context of a single test.